### PR TITLE
Add support for all authentication types, introduce JSON schema

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

File: v2/googlecloud-to-neo4j/src/main/resources/schemas/connection.schema.json [hunk 1]
```diff
@@ -0,0 +1,147 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://cloud.google.com/dataflow/googlecloud-to-neo4j/connection.schema.json",
```
Comment (fbiville) [inline v2/googlecloud-to-neo4j/src/main/resources/schemas/connection.schema.json:3]: not sure what I should put here @bvolpato

Comment (bvolpato) [reply v2/googlecloud-to-neo4j/src/main/resources/schemas/connection.schema.json:3]: We're deferring the ownership of the schema to Neo4j (and linking https://neo4j.com/docs/dataflow-bigquery/job-specification in our docs, for example)

Can you host this somewhere publicly?

Comment (fbiville) [reply v2/googlecloud-to-neo4j/src/main/resources/schemas/connection.schema.json:3]: Let me come back to you on that.

**Comment by fbiville** [comment]
@bvolpato rebased and ready to review

**Comment by fbiville** [comment]
@bvolpato the conflict has been fixed and the PR is now rebased again and ready for review

**Review by bvolpato — APPROVED** [review]

**Comment by fbiville** [comment]
@bvolpato the new conflict has been fixed and the PR has been rebased again

**Pull request merged**